azavea/iow-boundary-tool
# Reference Image Layer

## Overview

This PR add the reference image layer to the draw page. The layer is based on [Leaflet.DistortableImage](https://github.com/publiclab/Leaflet.DistortableImage). 

Closes #42 

### Demo

https://user-images.githubusercontent.com/8356789/186715207-8cc6c250-eceb-48b4-8e76-9a68aabf27a2.mov

### Notes

Updating the layer on state change was causing the image to become unselected. A simple work around for this is to remove the state data from the `useEffect` dependencies. This effectively means that the state is stored in two places: the distortable layer's internal state, and the redux global store. This is less than ideal and should be given future consideration.

## Testing Instructions
- Go to draw page
- Reference layer should be visible
- Toggle between each of the layer tools and test their functionality
- Turn off the reference layer and turn it back on, the image should disappear and appear in the same location with the same transformations

## Checklist
